### DISCUSSION TITLE
How to search by URL of remote toot using /api/v2/search?

### DISCUSSION THREAD

**Discussion by ShredderWalnut** [mastodon/mastodon#17583] [Q&A] [answered]
(This is a follow-up question of https://github.com/mastodon/mastodon/discussions/17574. Since I consider the two questions unrelated enough, I’m opening a new post.)

Hello,

My objective is to input the remote URL of a toot and retrieve its ID as seen from my Mastodon instance. I was instructed in the linked discussion above to use `/api/v2/search` for the purpose, but I can’t figure out how. Here’s part of the `GET` request in JavaScript:

```language-json
{
    "url": endpoint,
    "method": "GET",
    "parameters": {
        "q": "query"
    },
    "headers": {
        "Authorization": "Bearer " + token
    }
}
````
In the `"q": "query"` key, when I set the value to, say, `hello world`, the API returns a long list of search results as expected. But when I set the value to the URL of a toot, the API returns zero search results, i.e.

```language-json
{
    "accounts": [],
    "statuses": [],
    "hashtags": []
}
```
I tested with both toots on my local instance and remote instances and the results are the same.

Am I doing wrong by using the toot URL as the search query? Thank you in advance for your help!

**Comment by Gargron** [comment, marked as answer]
I think you need to also pass a `resolve=true` param.

**Reply by ShredderWalnut** [reply]
Hello Eugen,

Thanks, that indeed solved the issue. I overlooked all parameters marked as `optional` in the docs, so I missed this.
